src/MassTransit/Transports/BaseHost.cs

```diff
@@ -111,6 +126,8 @@ public async Task Stop(CancellationToken cancellationToken)
         {
             LogContext.Current = _hostConfiguration.LogContext;
 
+            await _riderCollection.Stop(cancellationToken).ConfigureAwait(false);
+
             await ReceiveEndpoints.Stop(cancellationToken).ConfigureAwait(false);
 
             foreach (var agent in GetAgentHandles())
```

Comment:
This is where the producers would be stopped and all connections to kafka/event-hub severed.